packages/flutter_tools/lib/src/android/deferred_components_setup_validator.dart

```diff
@@ -0,0 +1,723 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:meta/meta.dart';
+import 'package:yaml/yaml.dart';
+import 'package:xml/xml.dart';
+
+import '../base/common.dart';
+import '../base/deferred_component.dart';
+import '../base/file_system.dart';
+import '../base/terminal.dart';
+import '../build_system/build_system.dart';
+import '../build_system/depfile.dart';
+import '../globals.dart' as globals;
+import '../project.dart';
+import '../template.dart';
+
+/// A class to configure and run deferred component setup verification checks and tasks.
+///
+/// Once constructed, checks and tasks can be executed by calling the respective methods.
+/// The results of the checks are stored internally and can be displayed to the user by
+/// calling [displayResults].
+class DeferredComponentsSetupValidator {
+  DeferredComponentsSetupValidator(this.env, {
+    this.exitOnFail = true,
+    String title,
+    Directory templatesDir,
```

Comment:
Should this parameter be `@required`?

Reply:
This property can be null. If null, it will just use the default templates directory. I can document this.